pkgs/build-support/trivial-builders/default.nix

```diff
@@ -625,18 +625,29 @@ rec {
 
   # Docs in doc/build-helpers/trivial-build-helpers.chapter.md
   # See https://nixos.org/manual/nixpkgs/unstable/#trivial-builder-writeReferencesToFile
-  writeReferencesToFile = path: runCommand "runtime-deps"
+  # TODO: Convert to throw after Nixpkgs 24.05 branch-off.
+  writeReferencesToFile = (if config.allowAliases then lib.warn else throw)
+    "writeReferencesToFile is deprecated in favour of writeReferenceClosureToFile"
+    (path: writeReferenceClosureToFile [ path ]);
+
+  # Docs in doc/build-helpers/trivial-build-helpers.chapter.md
+  # See https://nixos.org/manual/nixpkgs/unstable/#trivial-builder-writeReferenceClosureToFile
+  writeReferenceClosureToFile = paths: runCommand "runtime-deps"
     {
-      exportReferencesGraph = [ "graph" path ];
+      # Get the cleaner exportReferencesGraph interface
+      __structuredAttrs = true;
+      exportReferencesGraph.graph = lib.pipe paths [
+        # Filter out common exceptions
+        (lib.filter (p: p != null && p != ""))
+        # String interpolation
+        (map (p: "${p}"))
+        # Sort paths to avoid unnecessary rebuild
+        (lib.sort lib.lessThan)
+      ];
+      nativeBuildInputs = [ jq ];
     }
     ''
-      touch $out
-      while read path; do
-        echo $path >> $out
-        read dummy
-        read nrRefs
-        for ((i = 0; i < nrRefs; i++)); do read ref; done
-      done < graph
+      jq -r ".graph | sort_by(.path) | .[].path" "$NIX_ATTRS_JSON_FILE" > "$out"
```

Comment:
Shaving the yak: `jq -r ".graph | map(.path) | sort_by(.) | .[]"`

Reply:
Now it's `jq -r ".graph | map(.path) | sort | .[]"`